src/main/java/de/danielbechler/diff/identity/IdentityStrategyResolver.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2014 Daniel Bechler
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.danielbechler.diff.identity;
+
+import de.danielbechler.diff.node.DiffNode;
+
+public interface IdentityStrategyResolver {
+	IdentityStrategy resolveIdentityStrategy(DiffNode node);
+	IdentityStrategy resolveByCollectionElement(Object collectionElement);
```

Comment:
That's the one I don't particularly need. :)
Included if someone needs across the board. Not very performant in its current implementation because it gets resolved to each collection element.